docs/dotnet.rst

```diff
@@ -0,0 +1,134 @@
+^^^^^^^^^^^^^^^^^^^^
+
+A few placeholders have been used in the steps below, while going through the steps replace these with the ones respective to your project:
+
+- ``<app-id>``: The name of your Flatpak, see :ref:`conventions:application ids`.
+- ``<app-name>``: The name of the root folder of your app repository
+- ``<project-name>``: The name of your ``.csproj`` file
+- ``<git-url>``: The URL to the git repository of the project
+- ``<git-tag>``: The numbered release to build from, in the form of a Git `Tag <https://git-scm.com/book/en/v2/Git-Basics-Tagging>`_.
+
+3.  Create a new folder somewhere different from your existing project
+
+4.  Create a YAML file titled ``<app-id>.yaml`` with the following example template, replacing the placeholders with the appropriate information. (Note: If your project file lives in a subfolder, be sure to include that in the build paths in this file and subsequent commands as well.): 
+
+.. code-block:: yaml
+
+  id: <app-id>
+  runtime: org.freedesktop.Platform
+  runtime-version: '23.08'
+  sdk: org.freedesktop.Sdk
+  sdk-extensions:
+    - org.freedesktop.Sdk.Extension.dotnet8
+  build-options:
+    prepend-path: "/usr/lib/sdk/dotnet8/bin"
+    append-ld-library-path: "/usr/lib/sdk/dotnet8/lib"
+    prepend-pkg-config-path: "/usr/lib/sdk/dotnet8/lib/pkgconfig"
+
+  command: <project-name>
+
+  finish-args:
+    - --device=dri
+    # TODO: Replace this with wayland and fallback-x11 once Wayland support
+    #       becomes available:
+    #       https://github.com/AvaloniaUI/Avalonia/pull/8003
+    - --socket=x11
+    - --share=ipc
+    - --env=DOTNET_ROOT=/app/lib/dotnet
+
+  modules:
+    - name: dotnet
+      buildsystem: simple
+      build-commands:
+      - /usr/lib/sdk/dotnet8/bin/install.sh
+
+    - name: <app-name>
+      buildsystem: simple
+      sources:
+        - type: git
+          url: <git-url>
+          tag: <git-tag>
+        - ./nuget-sources.json
+      build-commands:
+        - dotnet publish <project-name>.csproj -c Release --no-self-contained --source ./nuget-sources
+        - mkdir -p ${FLATPAK_DEST}/bin
+        - cp -r bin/Release/net8.0/publish/* ${FLATPAK_DEST}/bin
+
+.. note::
+
+    For providing access to other things such as the network or
+    filesystem, see :doc:`sandbox-permissions`
+
+5.  Copy and save the dotnet NuGet sources generator script
+    ``flatpak-dotnet-generator.py`` from the `Flatpak Builder Tools
+    repository <https://github.com/flatpak/flatpak-builder-tools>`_, to
+    the current folder, or run the following command to download it:
+
+.. code-block:: shell
+
+      wget https://raw.githubusercontent.com/flatpak/flatpak-builder-tools/master/dotnet/flatpak-dotnet-generator.py
+
+6.  Clone down your project repository to the folder
+
+.. code-block:: shell
+
+      git clone <git-url>
+
+7.  Install dependencies from Flathub
+
+.. code-block:: shell
+
+      flatpak-builder build-dir --user --repo=repo --install-deps-from=flathub <app-id>.yaml
```

Comment:
Delete this, step 9 already install deps

Reply:
The reason I kept this part in there is that the dependencies are required for the `flatpak-dotnet-generator.py` Python script on step 8 (which generates the `nuget-sources.json` file) as it executes a shell command for `dotnet restore` in the context of a Flatpak.

```python
subprocess.run([
    'flatpak', 'run',
    '--env=DOTNET_CLI_TELEMETRY_OPTOUT=true',
    '--env=DOTNET_SKIP_FIRST_TIME_EXPERIENCE=true',
    '--command=sh', f'--runtime=org.freedesktop.Sdk//{args.freedesktop}', '--share=network',
    '--filesystem=host', f'org.freedesktop.Sdk.Extension.dotnet{args.dotnet}//{args.freedesktop}', '-c',
    f'PATH="${{PATH}}:/usr/lib/sdk/dotnet{args.dotnet}/bin" LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/usr/lib/sdk/dotnet{args.dotnet}/lib" exec dotnet restore "$@"',
    '--', '--packages', tmp, args.project] + runtime_args)
```

Reply:
Ah ok. Then remove the `repo` argument and use `--download-only` otherwise this will start building

Reply:
Just updated!